src/traces/waterfall/hover.js

```diff
@@ -38,18 +38,44 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
     var size = (di.isSum) ? di.b + di.s : di.rawS;
 
     if(!di.isSum) {
-        // format delta numbers:
-        if(size > 0) {
-            point.extraText = formatNumber(size) + ' ' + DIRSYMBOL.increasing;
-        } else if(size < 0) {
-            point.extraText = '(' + (formatNumber(-size)) + ') ' + DIRSYMBOL.decreasing;
-        } else {
-            return;
+        point.initial = di.b + di.s - size;
+        point.delta = size;
+        point.final = point.initial + point.delta;
+    } else {
+        point.final = size;
+        point.initial = di.b;
+        point.delta = point.final - point.initial;
+    }
+
+    point.finalLabel = formatNumber(point.final);
+    point.deltaLabel = formatNumber(Math.abs(point.delta));
+    point.initialLabel = formatNumber(point.initial, 1);
+
+    var hoverinfo = di.hi || trace.hoverinfo;
+    var text = [];
+    if(hoverinfo && hoverinfo !== 'none' && hoverinfo !== 'skip') {
+        var isAll = (hoverinfo === 'all');
+        var parts = hoverinfo.split('+');
+
+        var hasFlag = function(flag) { return isAll || parts.indexOf(flag) !== -1; };
+
+        if(hasFlag('final') && point.finalLabel !== '') {
+            text.push('Final: ' + point.finalLabel);
+        }
+        if(hasFlag('delta') && point.deltaLabel !== '') {
+            if(size < 0) {
+                text.push('(' + point.deltaLabel + ') ' + DIRSYMBOL.decreasing);
```

Comment:
this here should be the `deltaLabel` value so that 

```
hoverinfo: 'delta'
```

and

```
hovertemplate: '%{delta}'
```

render the same.

Reply:
As for `Final: ` and `Initial: ` prefixes, they should **not** be part of `finalLabel` / `initalLabel`.

See example:

https://codepen.io/etpinard/pen/ewJNVJ

--------

In fact, those prefixes should only be there when there are more than 1 of 'initial', 'delta' and 'final' flags.

Reply:
In brief, _`xLabel`_ should have the default-formatted label text for _`x`_.

I'd argue that `if(size < 0)` logic for delta should be considered _formatting_, but the `DIRSMBOL` suffixes should **not**.

So in brief,

v = formatNumber(Math.abs(point.delta));
point.deltaLabel = size < 0 ? '(' + v + ')' : v;

// ...

    if(size < 0) {
                text.push(point.deltaLabel + DIRSYMBOL.decreasing))
      } else {
                text.push(point.deltaLabel +  DIRSYMBOL.increasing);
            }